### PR TITLE
Feature/external redirect

### PR DESCRIPTION
**Pull request by mr-nick17**
### What

Added functionality to allow a configurable external redirect from within Florence ie. user has a redirect param on the login screen and is redirected to said external florence path e.g. http://localhost:8081/florence/login?redirect=/wagtail/path. AKA ✅ [Add functionality to redirect externally from Florence login](https://jira.ons.gov.uk/browse/DIS-1708)
- Included a redirect screen that displays a 'loader' and link that will display desired URL for slow connections/issues with page loading
- Refactored existing code (tidy and remove superfluous code)
- Redirect is handler function that now calls additional functionality to handle (existing) internal and external redirects
- Added/updated unit tests

### How to review

Sense check
CI integration passes
Try running the changes and look at the clip for inspiration on what to test ⬇️ 
L👀k at the clip ⬇️  

#### Clip

https://github.com/user-attachments/assets/598ab9ed-6042-421a-80f0-b43727a3e66f

### Who can review

Frontend dev

### CODE REVIEW

File: src/app/utilities/redirect.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        const redirectPath = "/florence/collections";
+        handleRedirect(redirectPath);
+        expect(browserHistory.push).toHaveBeenCalledWith(redirectPath);
+    });
+
+    it("should call internalRedirect if redirectPath is not in allowedExternalPaths and known internal path", () => {
+        const redirectPath = "/florence/datasets";
+        handleRedirect(redirectPath);
+        expect(browserHistory.push).toHaveBeenCalledWith(redirectPath);
+    });
```
Comment (lindenmckenzie) [inline src/app/utilities/redirect.test.js:51]: Appreciate what you're getting at here but I'm not sure these statements accurately what's being executed.

File: src/app/utilities/redirect.js [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
+        return internalRedirect();
+    }
+
+    const config = window.getEnv();
+    const allowedExternalPaths = config.allowedExternalPaths;
+    if (allowedExternalPaths.some(path => redirectPath.startsWith(path))) {
+        return externalRedirect(redirectPath);
+    }
+
+    internalRedirect(redirectPath);
```
Comment (lindenmckenzie) [inline src/app/utilities/redirect.js:20]: Why no return?

**Review by lindenmckenzie — COMMENTED** [review]
Some very minor bits of feedback.

File: config/config.go [hunk 3]
```diff
@@ -40,14 +41,22 @@ func Get() (*Config, error) {
...
-		SharedConfig:               SharedConfig{EnableDatasetImport: true, EnableNewSignIn: false, EnableNewUpload: false, EnablePermissionsAPI: false, EnableCantabularJourney: false, EnableDataAdmin: true},
+		BindAddr:             ":8080",
+		APIRouterURL:         "http://localhost:23200",
+		APIRouterVersion:     "v1",
+		FrontendRouterURL:    "http://localhost:20000",
+		DatasetControllerURL: "http://localhost:24000",
+		TableRendererURL:     "http://localhost:23300",
+		DataAdminURL:         "http://localhost:29400/data-admin",
+		SharedConfig: SharedConfig{
+			AllowedExternalPaths:    []string{},
```
[the remark this answers is not in the document]
Comment (mr-nick17) [reply config/config.go:52]: Missed requirement 😉 

Not sure on the value of setting those values as defaults. IMHO the app should be vanilla and by default not have any default external paths as the app has no dependency on said external apps. Agreed it should still work without and having nothing as a default proves that. 

Happy to discuss further

File: src/app/utilities/redirect.test.js (hunk 1 above)
Comment (mr-nick17) [reply src/app/utilities/redirect.test.js:51]: The machines have to work on their test descriptions. We can discuss further on that

File: src/app/utilities/redirect.js (hunk 2 above)
Comment (mr-nick17) [reply src/app/utilities/redirect.js:20]: hmm git copilot deemed it unnecessary when I checked my refactoring, will put it back in

File: src/app/utilities/redirect.test.js (hunk 1 above)
Comment (mr-nick17) [reply src/app/utilities/redirect.test.js]: Have improved coverage and refactored how the descriptions are written 👍

File: src/app/utilities/redirect.js (hunk 2 above)
Comment (mr-nick17) [reply src/app/utilities/redirect.js]: Done

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**